app/controllers/api/json/imports_controller.rb

```diff
@@ -83,6 +83,46 @@ def create
     end
   end
 
+  # -------- Cancel a running import (using update PUT request) -------
+
+  def update
+    if !params[:item_queue_id].present?
+      render_jsonp({ errors: { imports: 'Missing queue ID in import cancellation request'} }, 400) and return
+    end
+
+    queue_id = params[:item_queue_id]
+
+    begin
+      # first verify that this job exists and user owns this job
+      import = DataImport.where(id: queue_id).first()
+
+      if import.nil?
+        render_jsonp({ errors: { imports: "Failed to cancel import. Job with ID #{queue_id} not found."} }, 400) and return
+      end
+
+      if import.user_id != current_user.id
+        render_jsonp({ errors: { imports: "Failed to cancel import. Job with ID #{queue_id} does not belong to current user."} }, 400) and return
```

Comment:
I think this should be combined with the check for `nil` above so that users can't discover other users' import IDs.